apps/remix-ide/src/app/plugins/code-format.ts

```diff
@@ -0,0 +1,77 @@
+'use strict'
+import { Plugin } from '@remixproject/engine'
+import prettier from 'prettier/standalone'
+import { Options } from 'prettier';
+import sol from './code-format/index'
+import * as ts from 'prettier/parser-typescript'
+import * as babel from 'prettier/parser-babel'
+import * as espree from 'prettier/parser-espree'
+import path from 'path'
+
+const profile = {
+    name: 'codeFormatter',
+    desciption: 'prettier plugin for Remix',
+    methods: ['format'],
+    events: [''],
+    version: '0.0.1'
+}
+
+export class CodeFormat extends Plugin {
+
+    constructor() {
+        super(profile)
+    }
+
+    async format() {
+        const file = await this.call('fileManager', 'getCurrentFile')
```

Comment:
I feel `format` should accept a `path` param, instead of getting the current file, if for instance one want to prettier entire project, or just prettier multiple files.

Reply:
sure